encoder/src/main/java/com/iabtcf/v2/encoder/TCModelBuilder.java

```diff
@@ -1,252 +1,61 @@
-		isServiceSpecific = coreString.isServiceSpecific();
-		useNonStandardStacks = coreString.isUseNonStandardStacks();
-		isPurposeOneTreatment = coreString.isPurposeOneTreatment();
-		publisherCountryCode = coreString.getPublisherCountryCode();
-		coreString.getAllOptedInSpecialFeatures().forEach(specialFeaturesOptInts::set);
-		coreString.getAllConsentedPurposes().forEach(purposesConsent::set);
-		coreString.getAllLegitimateInterestPurposes().forEach(purposesLITransparency::set);
-		coreString.getAllConsentedVendors().forEach(vendorConsents::add);
-		coreString.getAllLegitimateInterestVendors().forEach(vendorLegitimateInterests::add);
-		coreString.getAllPublisherRestrictions().forEach(pr -> {
-			RangeData rangeData = new RangeData();
-			pr.getAllVendors().forEach(rangeData::add);
-			publisherRestrictions.computeIfAbsent(pr.getPurpose(), p -> new EnumMap<>(RestrictionType.class))
-			                     .put(pr.getRestrictionType(), rangeData);
-
-		});
-
-		model.getOutOfBandConsent().ifPresent(oob -> {
-			oob.getAllDisclosedVendors().forEach(disclosedVendors::add);
-			oob.getAllAllowedVendors().forEach(allowedVendors::add);
-		});
-
-		model.getPublisherTC().ifPresent(ptc -> {
-			ptc.getAllConsentedPurposes().forEach(publisherPurposes::set);
-			ptc.getAllLegitimateInterestPurposes().forEach(publisherPurposesLI::set);
-			ptc.getAllConsentedCustomPurposes().forEach(publisherCustomPurposes::set);
-			ptc.getAllLegitimateInterestCustomPurposes().forEach(publisherCustomPurposesLI::set);
-		});
-	}
-
-	public TCModelBuilder version(final int val) {
-		version = val;
-		return this;
-	}
-
-	public TCModelBuilder consentRecordCreated(final Instant val) {
-		created = val;
-		return this;
-	}
-
-	public TCModelBuilder consentRecordLastUpdated(final Instant val) {
-		lastUpdated = val;
-		return this;
-	}
-
-	public TCModelBuilder consentManagerProviderId(final int val) {
-		cmpId = val;
-		return this;
-	}
-
-	public TCModelBuilder consentManagerProviderVersion(final int val) {
-		cmpVersion = val;
-		return this;
-	}
-
-	public TCModelBuilder consentScreen(final int val) {
-		consentScreen = val;
-		return this;
-	}
-
-	public TCModelBuilder consentLanguage(final String val) {
-		consentLanguage = val;
-		return this;
-	}
-
-	public TCModelBuilder vendorListVersion(final int val) {
-		vendorListVersion = val;
-		return this;
-	}
-
-	public TCModelBuilder policyVersion(final int val) {
-		policyVersion = val;
-		return this;
-	}
-
-	public TCModelBuilder isServiceSpecific(final boolean val) {
-		isServiceSpecific = val;
-		return this;
-	}
+	private CoreString.Builder coreStringBuilder;
+	private OutOfBandConsent.Builder outOfBandBuilder;
+	private PublisherTC.Builder publisherTCBuilder;
 
-	public TCModelBuilder useNonStandardStacks(final boolean val) {
-		useNonStandardStacks = val;
-		return this;
+	public TCModelBuilder() {
+		coreStringBuilder = CoreString.newBuilder();
+		outOfBandBuilder = OutOfBandConsent.newBuilder();
```

Comment:
outOfBandBuilder needs to be initialized to null in this constructor. Delete this line